docs/06-web-development.Rmd

```diff
@@ -101,7 +101,70 @@ There are much better tutorials on HTML than this section, such as those offered
 
 ## CSS
 
-CSS is a lot of fun to play with, but it can also easily steal your time.
+The Cascading Stylesheets (CSS) language is used to describe the look and formatting of documents written in HTML or XML. CSS is responsible for the visual style of your site. CSS is a lot of fun to play with, but it can also easily steal your time.
+
+In the Hugo framework (https://gohugo.io/tutorials/creating-a-new-theme/), CSS is one of the major “non-content” files that shapes the look and feel of your site (the others are images, JavaScript, and Hugo templates). What does the ["look and feel"](https://en.wikipedia.org/w/index.php?title=Look_and_feel&oldid=731052704) of a site mean? “Look” generally refers to static style components including, but not limited to:
+
+* Color palettes
+* Images
+* Layouts/margins
+* Fonts
+
+Whereas “feel” relates to dynamic components that the user interacts with like:
+
+* Dropdown menus
+* Buttons
+* Forms
+
+There are 3 ways to define styles. The first is inline with HTML. For example, this code:
+
+
+```html
+<text style="color:red">Marco!</text>
+Polo!
+```
+Would produce text that looks like this:
+<text style="color:red">Marco!</text>
+Polo!
+
+However, this method is generally not preferred for [numerous reasons](https://stackoverflow.com/questions/12013532/inline-style-tags-vs-inline-css-properties). 
+
+A second way is to internally define the CSS by placing a style section in your HTML:
+
+```html
+<html>
+<style> 
+#vehicle {
+    font-style: italic;
+}
+</style>
+<div id="book">
+  <p id="name">My name is Bananas Gorilla.</p>
+  <p id="aboutme">I wear lots of watches on both wrists.</p>
+  <p id="vehicle">I drive a bananamobile.</p>
+</div>
+</html>
+```
+
+In this example, the last line of text, <em>"I drive a bananamobile."</em>, is italicized. 
+
+The third method is the most popular because it is more flexible and the least repetitive, which is to define the CSS in an external file that is then referenced as a link in your HTML:
+
+```html
+<html>
+    <link rel="stylesheet" href="/css/style.css" />
+</html>
+```
+
+What goes inside the linked CSS document is essentially a list of rules (the same list could appear internally between the style tags, if you are using the second method). Each rule must include both a selector or group of selectors, and a declarations block within curly braces that contains one or more `property: value;` pairs. Here is the [general structure for a rule](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference):
+
+```css
+/* CSS pseudo-code */
+selectorlist {
+    property: value;
+    /* more property: value; pairs*/
+}
+```
 
 A few one-liner examples.
```

Comment:
I meant to show these examples at the end of this section. These look fairly simple but can produce amazing visual styles.